### PR TITLE
Calendar list

### PR DESCRIPTION
**Pull request by davidrapson**
Initial version of the calendar list view.

- Like  https://github.com/guardian/membership-frontend/pull/544 this is behind google auth
- Adds text label to view switch icon as @davidmcdowell and I felt it lacked enough context without the label.

Small screen is a little cramped but we only have desktop designs at the moment so haven't made too many assumptions for how that should look.

In discussion with @davidmcdowell as to whether we just have the list view as there is currently little distinction between grid and list (other than hiding the price in the grid). Putting both up so we can review them.

screen shot 2015-05-21 at 14 38 38

@mattandrews

### CODE REVIEW

File: frontend/app/model/Nav.scala [hunk 1]
```diff
@@ -10,13 +10,8 @@ object Nav {
...
 
-  def fetchNav(url: String) = {
-    val navigationURLs = navigation.find { navItem =>
-      // Return the parent item if URL is a subnav
-      navItem.href == url || navItem.subNav.map(_.href).contains(url)
-    }
-    navigationURLs
-  }
+  def fetchNav(url: String) =
+    navigation.find(navItem => url.contains(navItem.href))
```
Comment (davidrapson) [inline frontend/app/model/Nav.scala]: Previous logic prevented us showing the secondary nav in the list view as it doesn't need to be in the nav.

This version just matches against the url. In this case anything under `/whats-on` will return the what's on parent nav.

File: frontend/app/views/whatson/calendarList.scala.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+@(groupedEvents: scala.collection.SortedMap[org.joda.time.LocalDate, scala.collection.SortedMap[org.joda.time.LocalDate, Seq[model.RichEvent.RichEvent]]], pageInfo: model.PageInfo)
```
Comment (davidrapson) [inline frontend/app/views/whatson/calendarList.scala.html]: This gets pretty crazy when nested like this, I'm sure there's a terser way of modelling this, cc/ @rtyley

File: frontend/app/controllers/WhatsOn.scala [hunk 3]
```diff
@@ -21,6 +20,23 @@ trait WhatsOn extends Controller with ActivityTracking {
...
+  private def groupEventsByDayAndMonth(events: Seq[RichEvent]):  SortedMap[LocalDate, SortedMap[LocalDate, Seq[RichEvent]]] = {
+    val unsortedMap = groupEventsByDay(events).groupBy(_._1.withDayOfMonth(1))
+    SortedMap(unsortedMap.toSeq :_*)
+  }
+
+  private def groupEventsByMonth(events: Seq[RichEvent]): SortedMap[LocalDate, Seq[RichEvent]] = {
+    val unsortedMap = events.groupBy(_.start.toLocalDate.withDayOfMonth(1)).map {
+      case (monthDate, eventsForMonth) => monthDate -> eventsForMonth
+    }
+    SortedMap(unsortedMap.toSeq :_*)
```
Comment (mattandrews) [inline frontend/app/controllers/WhatsOn.scala]: Is this a fairly standard Scala thing? What's it doing?

File: frontend/app/views/whatson/calendarList.scala.html (hunk 2 above)
Comment (mattandrews) [inline frontend/app/views/whatson/calendarList.scala.html]: Any way to break it into multiple lines, at least?

**Comment by mattandrews** [comment]
:+1:

File: frontend/app/controllers/WhatsOn.scala (hunk 3 above)
Comment (davidrapson) [inline frontend/app/controllers/WhatsOn.scala]: Supposedly yes, the `:_*` is the way of converting a sequence into variable arguments. I guess a little like converting an array for use in `.call()` in JS.

**Comment by davidrapson** [comment]
Confirmed on PROD

screen shot 2015-05-22 at 11 18 44

**Pull request merged**